src/components/editor/metaform-editor-right-drawer-feature.tsx

```diff
@@ -554,6 +647,12 @@ const MetaformEditorRightDrawerFeatureComponent: FC<Props> = ({
     );
   };
 
+  useEffect(() => {
+    if (fieldIndex !== undefined) {
+      checkContextSettings();
+    }
```

Comment:
Move this conditional check into _checkContextSettings()_ method and only call the method itself from useEffect.